### ISSUE TITLE
EqualsAvoidNullCheck ignores left side variables with dot in it

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#1377]
I noticed EqualsAvoidNullCheck will not print errors on lines were the left side of the equals method has a "dot" in the variable like "this.field" or "MyClass.staticVariable". I don't see a reason these can't be swapped so the literals are on the left side and why the checker shouldn't flag them as errors. The source says it "cannot be improved" but I couldn't find any instance were the 2 couldn't be swapped.

Example Code:
````
package test;

public class TestClass3 {
	public static String staticTest = "";
	public String fieldTest = "";

	public void test() {
		String variable = "";

		if (variable.equals("123"))
			;

		if (TestClass3.staticTest.equals("123"))
			;

		if (this.fieldTest.equals("123"))
			;
	}
}
````

Check file:
````
<module name="Checker">
	<property name="charset" value="ASCII" />
	<property name="severity" value="error" />
	<property name="fileExtensions" value="java" />
	
	<module name="TreeWalker">
		<module name="EqualsAvoidNull" />
	</module>
</module>
````

checkstyle output:
````
Starting audit...
\TestClass3.java:10:36: String literal expressions should be on the left side of an equals comparison.
Audit done.
Checkstyle ends with 1 errors.
````

It only complains about the first if statement, while the other 2 go unnoticed. There is also nothing preventing them from being changed, without causing, issues to: "123".equals(outter.inner)

Source Involved:
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java#L165
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java#L173

**Comment by Vladlis** [comment]
Well, I also do not see the reason why the following example is ignored
```bash
vlad@vlad-N61PC-M2S ~/cli $ java -jar checkstyle-6.8.1-all.jar -c my_check.xml Test.java
Starting audit...
Audit done.

vlad@vlad-N61PC-M2S ~/cli $ cat Test.java
public class Test {
    String foo;
    static String bar;
    void method() {
        foo.equals(this.bar = "test");
    }
}

vlad@vlad-N61PC-M2S ~/cli $ cat my_check.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">


<module name = "Checker">
    <module name="TreeWalker">
         <module name="EqualsAvoidNullCheck"/>
    </module>
</module>
```

**Comment by Vladlis** [comment]
Looks like it's full of bugs
```bash
vlad@vlad-N61PC-M2S ~/cli $ java -jar checkstyle-6.8.1-all.jar -c my_check.xml Test.java
Starting audit...
Audit done.
vlad@vlad-N61PC-M2S ~/cli $ cat Test.java
public class Test {
    String foo;
    static String bar;
    void method() {
        (foo).equals("test");
    }
}
vlad@vlad-N61PC-M2S ~/cli $ cat my_check.xml
<module name = "Checker">
    <module name="TreeWalker">
         <module name="EqualsAvoidNullCheck"/>
    </module>
</module>
```

**Comment by romani** [comment]
> foo.equals(this.bar = "test");

that is nonsense, Checkstyle will never catch all craziness of human. NO need to validate that.
We do not put a violation on that code so it is good - too complicated for us.

> (foo).equals("test");

No need to validate that, as soon as extra symbols is removed (reported by other Check), we will catch that case.

**Comment by Vladlis** [comment]
@romani, ok, what about this case?
```bash
vlad@vlad-N61PC-M2S ~/cli $ java -jar checkstyle-6.8.1-all.jar -c my_check.xml Test.java
Starting audit...
Audit done.
vlad@vlad-N61PC-M2S ~/cli $ cat Test.java
class TestT {
    String foo;
    static String bar;
    void method() {
        (foo + bar).equals("test");
    }
}
vlad@vlad-N61PC-M2S ~/cli $ cat my_check.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">


<module name = "Checker">
    <module name="TreeWalker">
         <module name="EqualsAvoidNullCheck"/>
    </module>
</module>
```

**Comment by romani** [comment]
no need to check, you do not know type of "foo + bar", so you do not know what "equals" could be called.

**Comment by romani** [comment]
merged
